plugins/com.google.cloud.tools.eclipse.ui.util/src/com/google/cloud/tools/eclipse/ui/util/databinding/ProjectIdValidator.java

```diff
@@ -0,0 +1,34 @@
+package com.google.cloud.tools.eclipse.ui.util.databinding;
+
+import java.util.regex.Pattern;
+
+import org.eclipse.core.databinding.validation.IValidator;
+import org.eclipse.core.databinding.validation.ValidationStatus;
+import org.eclipse.core.runtime.IStatus;
+
+import com.google.cloud.tools.eclipse.ui.util.Messages;
+
+public class ProjectIdValidator implements IValidator {
+  // todo: this is different from .newproject.AppEngineProjectIdValidator
+  // But this reflects the reality when creating a project which says:
+  // Project ID must be between 6 and 30 characters.
```

Comment:
I think the other validator is based on internal AppId-related code.  The 6-30 characters comes from error message from the Cloud Console.